apps/protocol-frontend/src/components/EditContributionForm.tsx

```diff
@@ -89,6 +92,44 @@ const EditContributionForm = ({
     reset();
   };
 
+  const [, setIpfsUri] = useState('');
+  const [selectedFile, setSelectedFile] = useState<File | null>(null);
+  const fileInputField = useRef<HTMLInputElement>(null);
+  const [fileError, setFileError] = useState<string | null>(null);
+  const [isUploading, setIsUploading] = useState(false);
+
+  const handleFileUploadButtonClick = () => {
+    if (fileInputField.current !== null) {
+      fileInputField.current.click();
+    }
+  };
+  const handleImageSelect = async (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    const file = event?.target.files ? event.target.files[0] : null;
+    if (file !== null) {
+      setValue('proof', '');
+      setFileError(null);
+      if (file.size > MAX_FILE_UPLOAD_SIZE) {
+        setFileError(
+          'File size is too large. Please select something smaller than 5 MB.',
+        );
+        return;
+      }
+    }
+    setSelectedFile(file);
+  };
+
+  const handleIpfsUpload = async () => {
+    if (selectedFile) {
+      setIsUploading(true);
+      const ipfsImageUri = await uploadFileIpfs(selectedFile);
```

Comment:
It might be good to wrap this in a try and except and pop a toast if the file fails to be uploaded

Reply:
Oh good call, missed this earlier -- will add this in